core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -74,6 +75,18 @@ object Json {
     catch { case e: JsonProcessingException => Left(e) }
   }
 
+  /**
+    * Parse a JSON string into a JsonValue if possible. It returns an `Either` where `Left` will be an exception and
+    * `Right` is the `JsonValue`.
+    * @param input a JSON string to parse
+    * @return An `Either` which in case of `Left` means an exception and `Right` is the actual return value.
+    */
+  def tryParseFull(input: String): Either[JsonProcessingException, Option[JsonValue]] =
```

Comment:
It's a bit of an issue that this doesn't fallback while `parseFull` does. I would prefer both not to fallback, but to do that, we would also need https://github.com/apache/kafka/pull/4321.

Reply:
I'll ask @umesh9794 if I can take that over or he wants to finish it quickly and then we can proceed with this.